viewing_party/party.py

```diff
@@ -0,0 +1,255 @@
+    if host in subscriptions:
+        return True
+    else:
+        return False
+
+def get_friends_movies(user_data):
+    titles = set()
+    friends = user_data["friends"]
+    movies = []
+    for friend in friends:
+        for movie in friend["watched"]:
+            if movie["title"] not in titles: 
+                movies.append(movie)
+                titles.add(movie["title"])
+    return movies
+
+def get_available_recs(user_data):
+    friends_movies = get_friends_movies(user_data)
+    movie_recs = []
+
+    if len(friends_movies) == 0:
+        return []
+
+    titles_of_movies_user_watched = set()
+    for movie in user_data["watched"]:
+        titles_of_movies_user_watched.add(movie["title"])
+
+    for movie in friends_movies:
+        if movie["title"] not in titles_of_movies_user_watched:
+            if in_user_subscriptions(movie["host"], user_data["subscriptions"]):
+                movie_recs.append(movie)
+
+    return movie_recs
+
+#wave 5
+def get_new_rec_by_genre(user_data):
+    most_frequent_watched = get_most_watched_genre(user_data)
+    user_not_watched = get_friends_unique_watched(user_data)
+    recommended_movies = []
+
+    #######
+
+#def get_friends_not_watched(user_data):
+    if len(user_data["watched"]) == 0:
+        return []
+    
+    user_movies = []
+    for watched in user_data["watched"]:
+            user_movies.append(watched)
+    
+    friends_movies = []
+    for friends_data in user_data["friends"]:
+        for watched in friends_data["watched"]:
+                friends_movies.append(watched)
+
+    
+    recommend_movies = []
+    most_frequent_watched = get_most_watched_genre(user_data)
+    
+    for movie in friends_movies:
+        if movie not in user_movies and movie["genre"] == most_frequent_watched:
+            if movie in recommend_movies:
+                pass
+            else:
+                recommend_movies.append(movie)
+            
+
+    return recommend_movies
+    # if genre in user_has_not_watched:
+    #     if genre not in friends_movies:
+
+    # if most_frequent_watched and user_not_watched:
+    #     for movie in user_not_watched:
+    #         if movie["genre"] == most_frequent_watched:
+    #             recommended_movies.append(movie)
+        
+
+    # return recommended_movies
+    ##############
+
+def get_rec_from_favorites(user_data):
+    fav_movies = []
+    user_fav_movies = user_data["favorites"]
+    #user_not_watched = get_friends_unique_watched(user_data)
+    #friends_movie = get_friends_movies(user_data)
+    movies = get_unique_watched(user_data)
+
+    
+    for movie in movies: #user_fav_movies:
+        if movie in user_fav_movies:
+            fav_movies.append(movie)
+        #if len(user_fav_movies) == 0:
+            #if len(user_not_watched) == 0:
+                #return []
+    return fav_movies
+
+
+#     def get_available_recs(user_data):
+# #favorites = user_data["favorites"]
+#         return get_available_recs
+
```

Comment:
In future submissions, get rid of comments unless it describes code you'll refer to in the future.